### ISSUE TITLE
Enhance `TypesafeViewResponseFilter` to support redirects

### ISSUE THREAD

**Issue by erdlet** [mvc-toolbox/mvc-toolbox#25]
While using the typesafe views I thought it might be beneficial to have an additional flag `redirect` to support those
in a typesafe way too. So instead of having Controllers using `String` for redirects and some enum for templates, they could
use one (or maybe different) enums for both tasks.

So my proposal is to enhance the `@ViewName` annotation something like

```
public @interface ViewName {
  String value();
  boolean redirect() default false;
}
```

and the `TypesafeViewResponseFilter ` like this:

```
 @Override
  public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext)
      throws IOException {

     // ...

      if (viewName != null) {
+        if (viewName.redirect()) {
+          responseContext.setEntity(Redirect.to(viewName.value()).build());
+        } else {
            responseContext.setEntity(viewName.value());  
+        }
      }
    }
  }
```

**Comment by chkal** [comment]
That's an interesting one. A have a few thoughts I would like to share.

Conceptionally, `redirect=false` and `redirect=true` are very different things:

- `redirect=false` would mean that we refer to a view name, so basically a view packaged in the deployment artifact.
- `redirect=true` would mean that we refer to a URL the user sees in the browser's address bar.

Therefore, I wonder if we should encourage users to use two different enums for these two different concepts. Something like:

```java
public enum Views {

  @ViewName("user-list.jsp")
  USER_LIST,

}

@Redirect
public enum Redirects {

  @ViewName("/users")
  USER_LIST,

}
```

But in this case, using `@ViewName` is actually not correct from a naming perspective, as `/users` isn't a view name.

And another though. In many many cases, URLs aren't static (like `/users`), but contain path parameters (like `/users/{id}`). I guess if we want to support typesafe redirects, we need a way to specify values for path parameters as well. Maybe something like:

```java
return Redirect.to( Redirects.USER_DETAILS )
    .set("id", "chkal")
    .build();
```

To sum up, I guess this topic needs some further discussions?

**Comment by erdlet** [comment]
> To sum up, I guess this topic needs some further discussions?

Yes, indeed. Thanks for your thoughts, they're really valuable.

Now after thinking a little more about this I'm sure you're absolutely right with this: `Therefore, I wonder if we should encourage users to use two different enums for these two different concepts`

> But in this case, using `@ViewName` is actually not correct from a naming perspective, as /users isn't a view name.

+1. Maybe `@RedirectTarget` or something similar might be a better fit.

> In many many cases, URLs aren't static (like /users), but contain path parameters (like /users/{id}).

Yes, I thought a lot about those during the last days. I think it'd be helpful to extend the response filter for static redirects and the `Redirect` API like you've posted it above. This way we can ensure type-safety and provide a single location for storing redirects.
